src/library/libraryview.h

```diff
@@ -14,6 +14,7 @@ class LibraryView {
     virtual ~LibraryView() {};
 
     virtual void onShow() = 0;
+    virtual bool hasFocus() = 0;
```

Comment:
Wouldn't it make sense to provide QWidget::hasFocus() as a default implementation instead of declaring the function as pure virtual and let derived classes decide if they really need to override the default implementation? I see a lot of code duplication in derived classes that are overriding this function, always using the same implementation.